integrations/coinbase-integration/src/main/java/org/dash/wallet/integration/coinbase_integration/ui/CoinbaseConvertCryptoFragment.kt

```diff
@@ -261,6 +258,33 @@ class CoinbaseConvertCryptoFragment : Fragment(R.layout.fragment_coinbase_conver
         }
 
         monitorNetworkChanges()
+
+        lifecycleScope.launch {
+            sharedViewModel.baseIdForFaitModelCoinBase.collect { uiState ->
+                // New value received
+                when (uiState) {
+                    is BaseIdForFaitDataUIState.Success -> {
+                        uiState.baseIdForFaitDataList.let { list ->
+                            viewModel.setBaseIdForFaitModelCoinBase(list)
+                        }
+                    }
+
+                    is BaseIdForFaitDataUIState.LoadingState ->{
+                        if (uiState.isLoading) {
+                            showProgress(R.string.loading)
```

Comment:
This is from #1009. This code adds more loading dialogs while we're trying to get rid of them. Is it really necessary here? @hadia

Reply:
@Syn-McJ  we need to keep status if the user choose feature before all data related loaded

Reply:
Yes, but this code sets `_baseIdForFaitModelCoinBase` which is only needed when the swap trade is initiated, not before. So technically, we don't need a loading dialog for this until the user actually starts the conversion process, right?